cmd/weaver/load_uprobe.go

```diff
@@ -133,7 +151,15 @@ func loadUprobeAndBPFModule(traceContext *functionTraceContext, runtimeContext c
 		var valueString string
 		var outputValue outputArg
 		for {
+			// First sent values are process info
 			value := <-channel
+			output.ProcInfo = procInfo{}
+			err := output.ProcInfo.unmarshalBinary(value)
+			if err == nil {
+				// if err == nil value was proc info struct, else do
+				// not fetch next value
+				value = <-channel
+			}
```

Comment:
What's the logic here? When I run this I notice a handful of process info's getting dropped, but the argument data always appears to be accurate. I know perf buffers will drop records if the consumer isn't picking them up fast enough, do you know if that's whats happening here?

Reply:
I pasted output from running this on the tester here: https://gist.github.com/grantseltzer/dc86577772e5cc73c68178e6bba0aebf

It appears that the procInfo is only dropped if the function it's tracing has multiple arguments, which is super weird. I'm investigating today.

Reply:
When there is multiple arguments procInfo was being overwritten, I fixed it with latest commit... should be ok now